lib/calculate.h

```diff
@@ -62,14 +63,14 @@ R calculate(const std::string& s, const T& x, const T& y, bool* error = nullptr)
     case '*':
         return wrap(x * y);
     case '/':
-        if (isZero(y)) {
+        if (isZero(y) || (std::is_integral<T>::value && std::is_signed<T>::value && isEqual(y, T(-1)) && isEqual(x, std::numeric_limits<T>::min()))) { // cppcheck-suppress knownConditionTrueFalse
```

Comment:
Write `std::is_integral<T>{}` and `std::is_signed<T>{}` instead of `::value`.

Reply:
I would have used `std::is_integral_v<T>` if it was available, but yours works in C++11 and also "fixes" the FP 👍